rendermodules/materialize/schemas.py

```diff
@@ -0,0 +1,121 @@
+    maxLevel = Int(required=False, default=0, description=(
+        "maximum mipMapLevel to generate."))
+    fmt = Str(required=False, validator=validate.OneOf(['PNG', 'TIF', 'JPG']),
+              description=("image format to generate mipmaps -- "
+                           "PNG if not specified"))
+    maxOverviewWidthAndHeight = Int(required=False, description=(
+        "maximum pixel size for width or height of overview image.  "
+        "If excluded or 0, no overview generated."))
+    skipInterpolation = Boolean(required=False, description=(
+        "whether to skip interpolation (e.g. DMG data)"))
+    binaryMask = Boolean(required=False, description=(
+        "whether to use binary mask (e.g. DMG data)"))
+    label = Boolean(required=False, description=(
+        "whether to generate single color tile labels rather "
+        "than actual images"))
+    createIGrid = Boolean(required=False, description=(
+        "whther to create an IGrid file"))
+    forceGeneration = Boolean(required=False, description=(
+        "whether to regenerate existing tiles"))
+    renderGroup = Int(required=False, description=(
+        "index (1-n) identifying coarse portion of layer to render"))
+    numberOfRenderGroups = Int(required=False, description=(
+        "used in conjunction with renderGroup, total number of groups "
+        "being used"))
+
+
+class ZRangeParameters(argschema.schemas.DefaultSchema):
+    # TODO how does this work with zValues?
+    minZ = Int(required=False, description=("minimum Z integer"))
+    maxZ = Int(required=False, description=("maximum Z integer"))
+
+
+class RenderWebServiceParameters(argschema.schemas.DefaultSchema):
+    baseDataUrl = Str(required=False, description=(
+        "api endpoint url e.g. http://<host>[:port]/render-ws/v1"))
+    owner = Str(required=False, description=("owner of target collection"))
+    project = Str(required=False, description=("project fo target collection"))
+
+
+class RenderParametersRenderWebServiceParameters(RenderWebServiceParameters):
+    render = Nested(
+        RenderClientParameters, only=['owner', 'project', 'host', 'port'],
+        required=False)
+
+    @post_load
+    def validate_options(self, data):
+        # fill in with render parameters if they are defined
+        if data.get('owner') is None:
+            data['owner'] = data['render']['owner']
+        if data.get('project') is None:
+            data['project'] = data['render']['project']
+        if data.get('baseDataUrl') is None:
+            data['baseDataUrl'] = '{host}{port}/render-ws/v1'.format(
+                host=(data['render']['host']
+                      if data['render']['host'].startswith(
+                      'http') else 'http://{}'.format(data['render']['host'])),
+                port=('' if data['render']['port'] is None else ':{}'.format(
+                      data['render']['port'])))
+
+
+class SparkParameters(argschema.schemas.DefaultSchema):
```

Comment:
This is also duplicated I think

Reply:
This is the more general class of what is also implemented in the pointmatching.  Pointmatching currently implements this as part of a single schema class with module-specific parameters.  In a better world, parameters for spark modules would subclass something like this -- similar to what I have here.

Reply:
Okay.. lets make an issue for this and address it later, could consolidate and reduce codebase.